renderer/brave_wallet/brave_wallet_render_frame_observer.cc

```diff
@@ -91,9 +85,17 @@ void BraveWalletRenderFrameObserver::DidClearWindowObject() {
   v8::MicrotasksScope microtasks(isolate, context->GetMicrotaskQueue(),
                                  v8::MicrotasksScope::kDoNotRunMicrotasks);
 
-  if (dynamic_params.brave_use_native_ethereum_wallet &&
-      web_frame->GetDocument().IsDOMFeaturePolicyEnabled(context, "ethereum")) {
+  auto dynamic_params = get_dynamic_params_callback_.Run();
+  if (!dynamic_params.install_window_brave_ethereum_provider &&
+      !dynamic_params.install_window_ethereum_provider &&
+      !dynamic_params.brave_use_native_solana_wallet) {
+    return;
+  }
+
+  if (web_frame->GetDocument().IsDOMFeaturePolicyEnabled(context, "ethereum")) {
     JSEthereumProvider::Install(
+        dynamic_params.install_window_brave_ethereum_provider,
```

Comment:
we need early return if !install_window_brave_ethereum_provider && install_window_ethereum_provider which should not happen. We need a test case to guard it to make sure that doesn't happen by checking both `window.braveEthereum` and `window.ethereum` don't get injected.

Reply:
I've added a check before JSEthereumProvider::Install call

Reply:
Added, also added install_brave_ethereum_provider check here before JSEthereumProvider::Install call

Reply:
Added additional checks to tests "DoNotAttachIfMetaMaskInstalled" and "DoNotAttachIfNoWalletCreated" to check conditions when window.ethereum is not created while window.braveEthereum is created